### PR TITLE
Bug 1069617 - Put text inside anchor tags in music list view, and assign...

### PR DESCRIPTION
**Pull request by eeejay**
... aria roles.

### CODE REVIEW

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=41d006fa7d14)

File: apps/music/js/utils.js [hunk 1]
```diff
@@ -75,13 +77,13 @@ function createListElement(option, data, index, highlight) {
       a.dataset.keyRange = 'all';
       a.dataset.option = data.option;
 
-      li.appendChild(titleSpan);
+      a.appendChild(titleSpan);
```
Comment (jimporter) [inline apps/music/js/utils.js]: I'm a little concerned about this change; why is it needed?

File: apps/music/style/music.css [hunk 2]
```diff
@@ -1006,14 +1006,8 @@ html[dir="rtl"] li.list-item {
...
 li.list-item a {
   display: block;
-  color: #fff;
   text-decoration: none;
-  outline: none;
   margin: 0;
-  text-shadow: black 0.1rem 0.1rem;
-  white-space: nowrap;
-  overflow: hidden;
-  text-overflow: ellipsis;
```
Comment (jimporter) [inline apps/music/style/music.css]: Why are these removed? Are they no longer needed?

File: apps/music/js/utils.js (hunk 1 above)
Comment (eeejay) [inline apps/music/js/utils.js]: Currently, the label (`<span>`) and link (`<a>`) are siblings. The screen reader "sees" the labels, but they are inoperable because the listener expects the link to be the target. When the user swipes through the list items they hear "label foo", and then on the next item simply "link". This needs to be cleaned up and made logical. So in this patch we make the labels proper children of the link elements to give these lists a logical markup and allow the screen reader to navigate and operate it correctly.

File: apps/music/style/music.css (hunk 2 above)
Comment (eeejay) [inline apps/music/style/music.css]: I couldn't find anywhere in the application where there is a child `<a>` in a `.list-item`. So I re-appropriated that rule and cleaned it up for the case above.

Comment (dominickuo) [inline apps/music/style/music.css]: @eeejay If I recalled correctly, we used to have `<a>` in a list element but we tried to reduce the children counts in the DOM tree, so `<a>` was removed in some commit, and this might be the styles we forget to clean.

File: apps/music/js/utils.js (hunk 1 above)
Comment (dominickuo) [inline apps/music/js/utils.js]: Does the list elements also use `<a>` in the other apps? I remember why we use `<li>` was because the building blocks suggested the apps to use `<li>` for list elements.

Comment (dominickuo) [inline apps/music/js/utils.js]: Never mind, I think I misunderstood it :P

File: apps/music/js/ui/views/list_view.js [hunk 3]
```diff
@@ -95,6 +95,8 @@ var ListView = {
       headerLi = document.createElement('li');
       headerLi.className = 'list-header';
       headerLi.textContent = this.lastFirstLetter || '?';
+      headerLi.setAttribute('role', 'heading');
+      headerLi.setAttribute('aria-level', '2');
```
Comment (dominickuo) [inline apps/music/js/ui/views/list_view.js]: @eeejay When music app is in searching modes, like searching in tiles or lists, the search results will have something like the list headers(Artists, Albums or Tracks), do we need to also set role and aria-level for them?

Comment (eeejay) [inline apps/music/js/ui/views/list_view.js]: Do you mean the gaia-subheader? That needs to be given a role as well, but i think it is outside the scope of this bug.

Comment (dominickuo) [inline apps/music/js/ui/views/list_view.js]: @eeejay Yes, just thought that's something related to the list elements but we can do it another bug.

File: apps/music/index.html [hunk 4]
```diff
@@ -156,7 +156,7 @@ <h1 id="title-text" data-l10n-id="music"></h1>
           <span data-l10n-id="search-titles">TrackS</span>
             (<span class="search-result-count"></span>)
         </gaia-subheader>
-        <div class="search-results"></div>
+        <div class="search-results" role="listbox"></div>
```
Comment (dominickuo) [inline apps/music/index.html]: @eeejay There are two more elements use `search-results` class and we should also add `role="listbox"` to them?

Comment (eeejay) [inline apps/music/index.html]: Oops, good catch!

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=cebdd7199854)
